socorro/processor/signature_utilities.py

```diff
@@ -687,41 +687,32 @@ def _get_crashing_thread(self, processed_crash):
 
     #--------------------------------------------------------------------------
     def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-        result = super(SignatureRunWatchDog, self)._action(
-            raw_crash,
-            raw_dumps,
-            processed_crash,
-            processor_meta
-        )
         processed_crash['signature'] = (
             "shutdownhang | %s" % processed_crash['signature']
         )
-        return result
+        return True
 
 
 #==============================================================================
-class SignatureShutdownTimeout(SignatureGenerationRule):
+class SignatureShutdownTimeout(Rule):
     """Customize the signature for shutdown timeouts"""
 
     def version(self):
         return '1.0'
 
     def _predicate(self, raw_crash, raw_dumps, processed_crash, proc_meta):
-        return 'AsyncShutdownTimeout' in raw_crash
+        try:
+            return bool(raw_crash['AsyncShutdownTimeout'])
+        except KeyError:
+            return False
 
     def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
-        result = super(SignatureShutdownTimeout, self)._action(
-            raw_crash,
-            raw_dumps,
-            processed_crash,
-            processor_meta
-        )
         parts = ['AsyncShutdownTimeout']
         try:
-            shutdown_data = ujson.loads(raw_crash['AsyncShutdownTimeout'])
+            shutdown_data = json.loads(raw_crash['AsyncShutdownTimeout'])
```

Comment:
No, ujson is still the preferred tool. 
There was a time when ujson failed horribly with segfaults. But it was lucky because it was horribly broken raw crash JSON that weighed many gigabytes! Had ujson NOT failed those documents would have been saved (the processed crash would still be small) and cause excess disk usage and would cause problems inside caching etc. 

We used that moment to instead direct our attention to the crashes from breakpad and I think Ted was able to change something so it doesn't happen again. We also put in some code that prevents json documents from being too large in case this ever happens again. 

Lastly, at the time we discovered we ran an ancient version of ujson. We later upgraded. After upgrading I tested putting those Gb raw crashes through ujson and then it did work. 

I do suspect that there are a great deal of cases where the difference between ujson and regular json is 0.0001 seconds. :)